azkaban-common/src/main/java/azkaban/utils/ParamReplacement.java

```diff
@@ -0,0 +1,88 @@
+package azkaban.utils;
+
+import com.google.common.collect.Lists;
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+import org.joda.time.format.PeriodFormatter;
+import org.joda.time.format.PeriodFormatterBuilder;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ParamReplacement {
+  private static abstract class Param {
+    abstract String name();
+
+    abstract String replacement(String param);
+
+    String replace(String content) {
+      Pattern p = Pattern.compile("\\$\\{" + name() + "(?:|:.*?)}");
+      Matcher m = p.matcher(content);
+      String filledContent = "";
+      int lastIndex = 0;
+      while (m.find()) {
+        String g = m.group();
+        String replacement;
+        if (("${" + name() + "}").equals(g)) {
+          replacement = replacement(null);
+        } else {
+          String pstr =
+              org.apache.commons.lang.StringUtils.removeStart(g, "${" + name()
+                  + ":");
+          pstr = org.apache.commons.lang.StringUtils.removeEnd(pstr, "}");
+          replacement = replacement(pstr);
+        }
+        filledContent += content.substring(lastIndex, m.start()) + replacement;
+        lastIndex = m.end();
+      }
+      filledContent += content.substring(lastIndex);
+      return filledContent;
+    }
+  }
+
+  private static List<Param> params = Lists.<Param> newArrayList(
+  // time param, format: "${time:<plus period>,<time format>}", like
+  // "${time:-1day,yyyy-MM-dd HH:mm:ss}" will be replaced into like
+  // "2015-09-16 11:23:45", if current time is "2015-09-17 11:23:45"
+      new Param() {
+        PeriodFormatter periodFormatter = new PeriodFormatterBuilder()
+            .appendYears().appendSuffix("year").appendMonths()
+            .appendSuffix("month").appendDays().appendSuffix("day")
+            .appendHours().appendSuffix("hour").appendMinutes()
+            .appendSuffix("minute").appendSeconds().appendSuffix("second")
+            .toFormatter();
+
+        public String name() {
+          return "time";
+        }
+
+        public String replacement(String param) {
+          if (param == null) {
+            param = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX";
```

Comment:
Could make that configurable globally:
```java
resolvedProps.getString("time.replacement.default.format", "yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
```

So that I could for example set `time.replacement.default.format=yyyyMMdd` on the highest level and then just use `${time:-1day}` in some jobs to yield `20160203` etc.